gc.c

```diff
@@ -939,6 +939,9 @@ static const bool HEAP_PAGE_ALLOC_USE_MMAP = false;
 static bool heap_page_alloc_use_mmap;
 #endif
 
+#define RVALUE_AGE_BIT_COUNT 2
+#define RVALUE_AGE_BIT_MASK (~((bits_t)-1L << RVALUE_AGE_BIT_COUNT))
```

Comment:
This relies on the fact that most implementations use 2's complement for negative integers. The C standard does not enforce this so it would not work on implementations that don't use it.

From paragraph 2 of section 6.2.6.2 of the [C spec](https://www.open-std.org/jtc1/sc22/wg14/www/docs/n1124.pdf):

> If the sign bit is one, the value shall be modified in one of the following ways:
> — the corresponding value with sign bit 0 is negated (sign and magnitude);
> — the sign bit has the value −(2N ) (two’s complement);
> — the sign bit has the value −(2N − 1) (ones’ complement).

```suggestion
#define RVALUE_AGE_BIT_MASK ((1UL << RVALUE_AGE_BIT_COUNT) - 1)
```